snippets/javascript/next-ts.json

```diff
@@ -0,0 +1,172 @@
+  },
+  "import Next.js InferGetServerSidePropsType": {
+    "prefix": "ifgss",
+    "body": "import { InferGetServerSidePropsType } from 'next'",
+    "description": "Next.js InferGetServerSidePropsType import"
+  },
+  "use Next.js InferGetServerSidePropsType": {
+    "prefix": "ufgss",
+    "body": [
+      "function ${1:${TM_FILENAME_BASE}}({ data }: InferGetServerSidePropsType<typeof getServerSideProps>) {",
+      "\treturn $2",
+      "}"
+    ],
+    "description": "use InferGetServerSidePropsType snippet"
+  },
+  "import Next.js Head": {
+    "prefix": "imh",
+    "body": "import Head from 'next/head';",
+    "description": "Next.js Head import"
+  },
+  "import Next.js Link": {
+    "prefix": "iml",
+    "body": "import Link from 'next/link';",
+    "description": "Next.js Link import"
+  },
+  "Use Next.js Link": {
+    "prefix": "usl",
+    "body": ["<Link href=\"$1\">", "\t<a>", "\t\t$0", "\t</a>", "</Link>"],
+    "description": "Next.js Link Tag with <a>"
+  },
+  "Use Next.js Link With Pathname": {
+    "prefix": "uslp",
+    "body": [
+      "<Link href={{ pathname: \"$1\", query: { queryName: queryValue } }}>",
+      "\t<a>",
+      "\t\t$0",
+      "\t</a>",
+      "</Link>"
+    ],
+    "description": "Next.js Link with Pathname"
+  },
+  "Use Next.js LinkTagWithDynmicRoute": {
+    "prefix": "usld",
+    "body": [
+      "<Link href=\"/[$1]\" as={`/${2:SLUG_NAME}`}>",
+      "\t<a>",
+      "\t\t$0",
+      "\t</a>",
+      "</Link>"
+    ],
+    "description": "Next.js Link Tag with Dynamic Route"
+  },
+  "importNextRouter": {
+    "prefix": "imro",
+    "body": "import Router from 'next/router';",
+    "description": "Next.js Router import"
+  },
+  "Next.js Router from useRouter": {
+    "prefix": "usro",
+    "body": "const router = useRouter();",
+    "description": "Declare Next.js Router from useRouter"
+  },
+  "Next.js query param from useRouter": {
+    "prefix": "nroq",
+    "body": "const { $1 } = router.query;",
+    "description": "Destructure Next.js query param from Router from useRouter"
+  },
+  "importNextRouterWithRouter": {
+    "prefix": "imrow",
+    "body": "import Router, { withRouter } from 'next/router';",
+    "description": "Next.js Router and { withRouter } import"
+  },
+  "importNextUseRouter": {
+    "prefix": "imuro",
+    "body": "import { useRouter } from 'next/router';",
+    "description": "Next.js { useRouter } import"
+  },
+  "Import Next Image component": {
+    "prefix": "imni",
+    "body": "import Image from 'next/image';",
+    "description": "Next.js 10 Image component import"
+  },
+  "Use sized Next Image component": {
+    "prefix": "usim",
+    "body": "<Image alt=\"$1\" src=\"$1\" width=\"$1\" height=\"$1\" />",
+    "description": "Use sized Next Image component"
+  },
+  "Use unsized Next Image component": {
+    "prefix": "usimu",
+    "body": "<Image alt=\"$1\" src=\"$1\" unsized=\"$1\" />",
+    "description": "Use sized Next Image component"
+  }
```

Comment:
You can remove these lines as they are already present in the `next.json` file.

Reply:
okay. i understand

Reply:
Thanks for the quick changes :+1: